lib/app-layout/index.tsx

```diff
@@ -144,7 +144,9 @@ export class AppLayout extends Component<Props> {
 const mapStateToProps: S.MapState<StateProps> = (state) => ({
   hasRevisions:
     state.ui.showRevisions && state.data.noteRevisions.has(state.ui.openedNote),
-  keyboardShortcutsAreOpen: state.ui.dialogs.includes('KEYBINDINGS'),
+  keyboardShortcutsAreOpen:
+    state.ui.dialogs.find((element) => element.type === 'KEYBINDINGS') !==
+    undefined,
```

Comment:
simply because this is a big complicated and specific to the dialogs reducer I wonder if we should add a selector in `selectors.ts` to make it clearer in the normal code:

```ts
export const isDialogOpen = (state: S.State, name: T.DialogType) =>
	state.ui.dialogs.find(({type}) => type === name) !== undefined
```

```ts
import * from selectors;

…

keyboardShortcutsAreOpen: selectors.isDialogOpen('KEYBINDINGS'),
```

normally I wouldn't encourage writing a selector or function for a single use like this but that's kind of quirky code _necessary_ to get the right answer and moving it into its own testable place might be helpful

Reply:
This certainly makes it cleaner to read when in `app-layout.tsx` I end up with a type error for `'KEYBINDINGS'` though `Argument of type 'string' is not assignable to parameter of type 'DialogType'.ts(2345)`

There are a couple of other places where I've run into similar as well. I'm not sure how to best deal with it. Another example is in the [reducer](https://github.com/Automattic/simplenote-electron/blob/add/delete-tag-confirmation-middleware/lib/state/ui/reducer.ts#L108). The last `tagName` there gives the error 
`Property 'tagName' does not exist on type 'Action<"SHOW_DIALOG", { name: "ABOUT" | "BETA-WARNING" | "CLOSE-WINDOW-CONFIRMATION" | "IMPORT" | "KEYBINDINGS" | "LOGOUT-CONFIRMATION" | "SETTINGS" | "SHARE" | "TRASH-TAG-CONFIRMATION"; data: object; }>'.ts(2339)`

It does all work though, any ideas or pointers you have?

Reply:
well my code examples are rarely correct or without bugs. `T.DialogType` was updated in this PR and I was reusing the old type. you can get just the `type` name by using `T.DialogType['type']`

Reply:
Of course, I totally missed that.